### DISCUSSION TITLE
Could you please have a example on how to use this with all layout coming from CMS? even the component name?

### DISCUSSION THREAD

**Discussion by patrickyue** [huang-julien/nuxt-runtime-compiler#45] [Q&A]
Hi, thanks for the module. it really helps 
But I still have a small qq here
I am using this code :

```
page data comes form CMS, and props to the PageTemplate component

./components/PageTemplate.vue

<script lang="ts">
  export default defineComponent({
    props: ['pageTemplate'],
     /**
     * most of the time, vue compiler need at least a VNode, use h() to render the component
     */
    render () {
      return h({
        props: ['pageTemplate'],
        template: this.pageTemplate
      })
    }
  })
</script>
```
I can see  the page without style, but I get this...
![image](https://user-images.githubusercontent.com/8323205/219797959-05f58691-29f8-4cf9-948b-fe3489cfdab4.png)
<navbar> and some other components are not HTML code, just as <navbar></navbar> inside HTML
I tried add components like this, but no luck

      return h({
        props: ['pageTemplate'],
        template: this.pageTemplate,
        components: {
          NavBar
      )}
so may I know if I miss something?

thanks in advance

**Comment by huang-julien** [comment]
Hi :wave: that's a bit strange since this component looks good to me. I could'nt reproduce the issue. Do you have a small reproduction ? Like a stackblitz or a repo ?
This is the `PageTemplate.vue` file i've been using to try to reproduce the issue:
````vue
<script lang="ts">
import { JustSomeComponent } from "#components"
  export default defineComponent({
    props: ['pageTemplate'],
     /**
     * most of the time, vue compiler need at least a VNode, use h() to render the component
     */
    render () {
      return h({
        props: ['pageTemplate'],
        template: this.pageTemplate,
        components: {
            JustSomeComponent
        }
      })
    }
  })
</script>
`````
and this the `page/index.vue` file 
```````ts

<template>
  <!-- Edit this file to play around with Nuxt but never commit changes! -->
  <div > 
    <PageTemplate page-template="<div><JustSomeComponent /></div>" />
  </div>
</template>

``````

**Reply by patrickyue** [reply]
Hi, huang
I think I fixed the issue, as I only have script tag in that component
actually we need 
<template>
    <component :is="renderer"></component>
</template>

Thanks, it works now

**Reply by huang-julien** [reply]
Happy to hear this ! But i didn't understand how did you solve this

**Reply by patrickyue** [reply]
Hi, Huang
I can explain a bit haha,
the previous issue is that inside my pageTemplate.vue, I only have <script> shown above there, nothing else. So it shows me "[Vue warn]: Component <Anonymous> is missing template or render function.  ". I solved it by adding that template with dynamic component and binding with that rendered function stuff. 

TBH, I also found that auto-import does not work with this run time renderer function, even you have A inside /components folder, but I also found that importing those components solves the issue.

for ex:
```
./pageTemplate.vue
<script lang="ts" setup>
import A from './components/A.vue'    
(without this line,  component A will not work, it will stay <A></A> inside the HTML.)

const props = defineProps<{
  template?: string
}>();

const renderer = computed(() => () => h({
  template: props.template,
  components: {
     A
  },
}))
</script>

<template>
  <component :is="renderer"></component>
</template>
```

Do you know if it is possible to let it support auto import, any suggestions?
Thanks

**Reply by huang-julien** [reply]
It is not possible to support auto-import saddly since auto-import is a build time plugin, so you have to know which component will be used by your components.

**Reply by patrickyue** [reply]
Good day, huang
I have a qq here for the new merge on this module
do u know what changes should we do once upgrade to nuxt@3.4.0+
looks like I don't need to install that nuxt compiler module, just remove it?

thanks

**Reply by huang-julien** [reply]
indeed, you can now use `experimental.runtimeVueCompiler`

**Reply by huang-julien** [reply]
btw it will change to `vue.runtimeCompiler` in 3.5

**Reply by patrickyue** [reply]
do you know where to find some code snippets for this ? 
thanks

**Reply by huang-julien** [reply]
for what exactly ? do you mean to enable the full bundle of vue ?
